src/wallet/wallet2.cpp

```diff
@@ -6858,8 +6858,12 @@ std::vector<wallet2::pending_tx> wallet2::create_transactions_2(std::vector<cryp
 		}
 		else
 		{
-			while(!dsts.empty() && dsts[0].amount <= available_amount && estimate_rct_tx_size(tx.selected_transfers.size(), fake_outs_count, tx.dsts.size(), bulletproof) < TX_SIZE_TARGET(upper_transaction_size_limit))
+			while(!dsts.empty() && dsts[0].amount <= available_amount && 
+				estimate_rct_tx_size(tx.selected_transfers.size(), fake_outs_count, tx.dsts.size(), bulletproof) < TX_SIZE_TARGET(upper_transaction_size_limit))
 			{
+				if(bulletproof && tx.dsts.size() >= cryptonote::common_config::BULLETPROOF_MAX_OUTPUTS-1)
```

Comment:
please recheck if this must be `>` and later on line `6879` `<=`

Reply:
Nope. We need one free slot for change.

Reply:
But since you already reduced the max outputs by one and use `>=` this means we use only 15 outputs.

Reply:
That's correct. You need one free slot for change.